### ISSUE TITLE
PrintToPdfAsync always hangs for 59 seconds before completing successfully

### ISSUE THREAD

**Issue by stevencohn** [MicrosoftEdge/WebView2Feedback#1974]
**Description**

I have a proof of concept written in LINQPad 6 like this:

```cs
var wv = new WebView2();
await wv.EnsureCoreWebView2Async();

wv.Source = new Uri("https://github.com/stevencohn/onemore");

wv.NavigationCompleted += async (sender, args) =>
{
	var watch = new Stopwatch();
	watch.Start();
	var path = Path.Combine(Path.GetTempPath(), Path.GetFileNameWithoutExtension(Path.GetRandomFileName()) + ".pdf");
	// *** this works but it is terribly slow!
	path.Dump("rendering pdf");
	var ok = await wv.CoreWebView2.PrintToPdfAsync(path);
	ok.Dump(path);
	watch.Elapsed.Dump();
	// shut down webview2 process
	wv.Dispose();
};
```

It completes successfully but every run takes just a little over 59 seconds according to the StopWatch report. Why?

**Version**
WebView2: 96.0.1054.34 WinForms component
Nuget: Microsoft.Web.WebView2 1.0.1054.31
OS: Win11

**Update**
I noticed that if I remove the call to Dispose that the first run will take 59 seconds but all subsequent runs will take about 0.5s to render a new PDF. 

Also, during those 59 seconds, it doesn't seem like anything is happening - no CPU, memory, or disk activity from any related process.

And I wrote a simple WinForms app that demonstrates exactly the same behavior so it's not something to do with LINQPad specifically

[AB#37224152](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/37224152)

**Comment by stevencohn** [comment]
So this isn't a solution but I discovered that if I introduce a Task.Delay(2000) before calling PrintToPdfAsync then it works OK. 

This delay must provide enough breathing room for some initialization beyond what is gated by both EnsureCoreWebView2Async and NavigationCompleted. But I haven't figured out what that might be.

**Comment by mikeduglas** [comment]
I see similar behaviour: sometimes PdfToPrint takes too long, but not every time. For example, if I print same page 10 times, 7 times it prints fast, 3 times it prints with long delay.

**Comment by champnic** [comment]
Thanks for the bug report, I've added it to our backlog to take a look.

**Comment by XHighIntell** [comment]
I tried the older version 1.0.1020.30, they are the same, adding delay 2000ms also helps.

**Comment by wilva** [comment]
Not sure if it helps, but when having this issue and not adding the delay it also triggers a ProcessFailed event.
The ProcessFailed event gets triggered with a processFailedKind "Render Process Unresponsive" and reason "Unresponsive".

**Comment by DanCooperstock** [comment]
The delay works for me too, both in a C# program using WebView2 directly, and in a program using AntView (an ActiveX cover for WebView2, which I'm using in PowerBuilder).

**Comment by DanCooperstock** [comment]
Is there an intent to fix this problem?

**Comment by vickiez** [comment]
Hi all, we recently made a fix, which should be available starting version 104.0.1277.0. Can you test with the latest Canary and confirm that the issue is resolved?
